keras_cv/models/resnet_v1.py

```diff
@@ -0,0 +1,357 @@
+# Copyright 2022 The KerasCV Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""ResNet models for Keras.
+Reference:
+  - [Deep Residual Learning for Image Recognition](https://arxiv.org/abs/1512.03385) (CVPR 2015)
+  - [Based on the original keras.applications ResNet](https://github.com/keras-team/keras/blob/master/keras/applications/resnet.py)
+"""
+
+import tensorflow as tf
+from tensorflow.keras import backend
+from tensorflow.keras import layers
+
+MODEL_CONFIGS = {
+    "ResNet50": {
+        "stackwise_filters": [64, 128, 256, 512],
+        "stackwise_blocks": [3, 4, 6, 3],
+        "stackwise_strides": [1, 2, 2, 2],
+    },
+    "ResNet101": {
+        "stackwise_filters": [64, 128, 256, 512],
+        "stackwise_blocks": [3, 4, 23, 3],
+        "stackwise_strides": [1, 2, 2, 2],
+    },
+    "ResNet152": {
+        "stackwise_filters": [64, 128, 256, 512],
+        "stackwise_blocks": [3, 8, 36, 3],
+        "stackwise_strides": [1, 2, 2, 2],
+    },
+}
```

Comment:
I think keeping the configs in the form of a dictionary helps provide a clear separation between the different parameters.